frame/contracts/src/lib.rs

```diff
@@ -1094,9 +1102,8 @@ impl<T: Config> Invokable<T> for InstantiateInput<T> {
 				&salt,
 				debug_message,
 			);
-			storage_deposit = storage_meter
-				.into_deposit(&origin)
-				.saturating_add(&StorageDeposit::Charge(extra_deposit));
```

Comment:
Why `StorageDeposit::Charge(extra_deposit)` is not being added here anymore?

Reply:
Yeah good point, initially I thought that it did not make much sense to keep it here since , the extra_deposit is 
not charged to the storage_deposit account but reserved on the uploading account. 

see here:
https://github.com/paritytech/substrate/blob/a5a63da2213d1f56c7e95fbb99578d6fda070a88/frame/contracts/src/lib.rs#L1075-L1079

the extra_deposit is used a few line above when we construct the meter to check if we have exceeded the storage_deposit_limit. This `storage_deposit` variable  mainly exists on the ContractResult so we can access it in test to validate some assertions.

but you are right we should probably add it back since this is supposed to reflect how much was consumed for the deposit.

Reply:
Correct. The deposit taken for the contract code upload is reserved [here](https://github.com/paritytech/substrate/blob/master/frame/contracts/src/wasm/code_cache.rs#L98). I think the idea behind returning the storage deposit taken\refunded as a result of a call is to provide this info back to the client, just like it's done with the ~~gas~~ weight used. So let alone it has a (slightly) different nature (reserving vs sending to a dedicated deposit account), I think we still should let a client know if an instantiate call required for it. 

Hence yeah, I agree this extra_deposit should be kept here.